tests/kernel/queue/src/test_queue_contexts.c

```diff
@@ -398,3 +408,39 @@ void test_multiple_queues(void)
 		tqueue_get(&queues[i]);
 	}
 }
+
+void user_access_queue_private_data(void *p1, void *p2, void *p3)
+{
+	/* try to access to private kernel data, will happen kernel oops */
+	k_queue_is_empty(&queue);
+
+	ret = TC_FAIL;
+}
+
+/**
+ * @brief Test access kernel object with private data using system call
+ *
+ * @details
+ * - When defining system calls, it is very important to ensure that
+ *   access to the API’s private data is done exclusively through system call
+ *   interfaces. Private kernel data should never be made available to user mode
+ *   threads directly. For example, the k_queue APIs were intentionally not made
+ *   available as they store bookkeeping information about the queue directly
+ *   in the queue buffers which are visible from user mode.
+ * - Current test makes user thread try to access private kernel data within
+ *   their associated data structures. Kernel will track that system call
+ *   access to these object with the kernel object permission system.
+ *   Current user thread doesn't have permission on it, trying to access
+ *   &pqueue kernel object will happen kernel oops, because current user
+ *   thread doesn't have permission on k_queue object with private kernel data.
+ *
+ * @ingroup kernel_memprotect_tests
+ */
```

Comment:
Could you re-write the doxygen comment as you did in #29528? We need them for generating test specification.

By the way, please also update the doxygen comments of your other PRs.

Reply:
I don't see that it is in Zephyr Community Guideline https://docs.zephyrproject.org/latest/development_process/documentation.html?highlight=doxygen

Reply:
We need it for test specification. Maybe I'm going to propose to use the new format of doxygen comment for all test cases. Of couse, this is another topic. 

For now, It's OK to get the PR approved/merged first.